ngmix/ksigmamom.py

```diff
@@ -0,0 +1,445 @@
+import logging
+
+import numpy as np
+import scipy.fft as fft
+
+from ngmix.observation import Observation
+from ngmix.moments import fwhm_to_sigma
+from ngmix.util import get_ratio_error
+
+
+logger = logging.getLogger(__name__)
+
+
+class KSigmaMom(object):
+    """Measure pre-PSF weighted real-sapce moments w/ the 'ksigma'
+    Fourier-space kernels from Bernstein et al., arXiv:1508.05655.
+
+    If the fwhm of the weight/kernel function is of similar size to the PSF or
+    smaller, then the object properties returned by this fitter will be very noisy.
+
+    Parameters
+    ----------
+    fwhm : float
+        This parameter is the approximate real-space FWHM of the kernel.
+    pad_factor : int, optional
+        The factor by which to pad the FFTs used for the image. Default is 4.
+    """
+    def __init__(self, fwhm, pad_factor=4):
+        self.fwhm = fwhm
+        self.pad_factor = pad_factor
+
+    def go(self, obs, return_kernels=False, no_psf=False):
+        """Measure the pre-PSF ksigma moments.
+
+        Parameters
+        ----------
+        obs : Observation
+            The observation to measure.
+        return_kernels : bool, optional
+            If True, return the kernels used for the flux and moments.
+            Defaults to False.
+        no_psf : bool, optional
+            If True, allow inputs without a PSF observation. Defaults to False
+            so that any input observation without a PSF will raise an error.
+
+        Returns
+        -------
+        result dictionary
+        """
+        if not isinstance(obs, Observation):
+            raise ValueError("input obs must be an Observation")
+
+        if not obs.has_psf() and not no_psf:
+            raise RuntimeError("The PSF must be set to measure a pre-PSF moment!")
+
+        if not no_psf:
+            psf_obs = obs.get_psf()
+
+            if psf_obs.jacobian.get_galsim_wcs() != obs.jacobian.get_galsim_wcs():
+                raise RuntimeError(
+                    "The PSF and observation must have the same WCS "
+                    "Jacobian for measuring pre-PSF moments."
+                )
+        else:
+            psf_obs = None
+
+        return self._meas_fourier_only(obs, psf_obs, return_kernels)
+
+    def _meas_fourier_only(self, obs, psf_obs, return_kernels):
```

Comment:
what does the `_only` part of the name imply here?

Reply:
Good catch! This is leftover from previous code versions. I renamed it.